### DISCUSSION TITLE
Record Level Multithreading

### DISCUSSION THREAD

**Discussion by alfa-f** [graphql-java/graphql-java#2320] [Q&A]
Hi,

I note from your article https://www.graphql-java.com/blog/threads/ that graphql-java does not do any thread management on its own as part of your design philosophy. 

I understand that I can feasibly resolve neighbouring entities or attributes in a multithreaded fashion by submitting the resolver code to a thread pool of my choosing.

However, is it possible to multithread at the _record_ level? ie. I return a list of 100 entities and the graphql-java engine can concurrently resolve these to Map<String, Object> and validate them in parallel? Our particular use case involves the subsciptions interface (ie. our resolver returns a finite stream of Entity). We noted when using Flux in the resolver layer with an elastic pool that the graphql-java execution still occurs on a single thread from the assigned pool.

**Comment by hareesh-r** [comment]
Hey did you find any answers to this ?

**Comment by bbakerman** [comment]
The graphql engine will never run N fields (the person above says "record" but I am unsure if that is what he meant) on different threads itself.

There is no advantage - given a list of fields `a,b,c,d` it will run them as a list of fields and dispatch each to a data loader - IF the data loader decides to have each be run in its own executor (aka thread) then great - but the engine will never thread fields itself
